### ISSUE TITLE
LDAP & Nested groups

### ISSUE THREAD

**Issue by seanzechman** [dalibo/ldap2pg#577]
I see in issue 348 - https://github.com/dalibo/ldap2pg/issues/348 it is mentioned that "As of now, ldap2pg only supports one level of sub-query for all values of an attributes" - has this support been added?  I tried several things to get nested group queries to work but no luck.

Basically the group we would like to create login roles from is nested in the group we have configured.

```
- description: "Query LDAP to create roles."
  ldap:
    base: CN=GROUP_of_GROUPS,OU=xxxxx,OU=xxxxx,DC=xxxxx,DC=com
    filter: "(objectClass=group)"
    join:
      member:
        filter: "(&(objectClass=user)(sAMAccountName=*))"
  grant:
    privilege: grant_personal_account_role
    role: '{member.sAMAccountName}'
  roles:
  - name: '{member.sAMAccountName}'
    options: LOGIN
    comment: 'From LDAP group {dn}'

```

And GROUP_of_GROUPS has groups with actual members in them that need to have login roles created.

**Comment by bersace** [comment]
hello. thanks for reaching

can you create server side views flattening groups structure ? this will performs better.

**Comment by seanzechman** [comment]
I was able to get this to work using the server side flatten group id

- description: "Query LDAP to create superusers."
  ldap:
    base: OU=xxxxx,OU=xxxxx,DC=xxxxx,DC=com
    filter: "(**memberOf:1.2.840.113556.1.4.1941:=**CN=GROUP_of_GROUPS,OU=xxxxx,OU=xxxxx,DC=xxxxx,DC=com)"
    attribute: sAMAccountName
  grant:
    privilege: grant_personal_account_role
    role: '{sAMAccountName}'
  roles:
  - name: '{sAMAccountName}'
    options: LOGIN SUPERUSER
    comment: '{Name} From LDAP group GROUP_of_GROUPS '

**Comment by bersace** [comment]
Thanks for the feedback.

note that `attribute` is already computed from `{...}` in rules.
See verbose logs with normalized configuration.

**Comment by seanzechman** [comment]
Hi bersace - Can you elaborate what you mean?  Are you stating that I do not need to specify attribute?  I used verbose a lot to debug my problems as it took a while to get a proper search criteria.

**Comment by bersace** [comment]
> Hi bersace - Can you elaborate what you mean? Are you stating that I do not need to specify attribute?

Yes. attribute is infered from templating.

``` diff
 - description: "Query LDAP to create superusers."
   ldap:
     base: OU=xxxxx,OU=xxxxx,DC=xxxxx,DC=com
     filter: "(memberOf:1.2.840.113556.1.4.1941:=CN=GROUP_of_GROUPS,OU=xxxxx,OU=xxxxx,DC=xxxxx,DC=com)"
-    attribute: sAMAccountName
   grant:
```

**Comment by ntap-jbo** [comment]
I've solved it like this, ldap2pg version 6.0.0 .

As this is was a test, it's not important that the base and filter are exposed.

```
- description: "Query LDAP to create roles."
  ldapsearch:
    base: DC=libvirt,DC=lan
    filter: "(&(memberOf:1.2.840.113556.1.4.1941:=CN=TestAll,DC=libvirt,DC=lan)(objectClass=person))"
  role:
  - name: '{sAMAccountName}'
    options: LOGIN
    comment: 'From LDAP group {dn}'
    parent: "writers"
    
    ```

**Comment by seanzechman** [comment]
We were able to work through this using the code I shared on Nov 9th comment - marking this completed
